### DISCUSSION TITLE
2 layman questions + 1 request

### DISCUSSION THREAD

**Discussion by joaosimuai** [stdword/logseq13-full-house-plugin#32] [Q&A] [answered]
Impressed by the plugin (I've used `pages-tagged-with`), but lost on its and .js handling, I'd like to ask:
- How to rename the displayed title, as in "`Pages with X containing`..."?
- How to change the direction (vertical → horizontal) of displayed blocks?

And **request**:
<details><summary>About "pages-tagged-with" code:</summary>
<p>

```typescript
``{
function getRelatedPages(relationName, relationValue) {
  const query = `[:find ?name
   :where
      [?p :block/original-name ?name]
      [?p :block/properties ?ps]
      [(get ?ps :${relationName}) ?pv]
      [(contains? ?pv "${relationValue}")]
  ]`
  const result = top.logseq.api.datascript_query(query)
  if (!result || result.length === 0)
      return null
  return result.flat()
}

const property = c.args.prop || 'tags'
const relation = c.page.name
const pages = getRelatedPages(property, relation)
console.log(pages)
if (!pages)
  return `There are no pages with property <b><i>${property}</i></b> and relation [[${relation}]]`
_}``
Pages with ``dev.toHTML(ref(property))`` containing ``dev.toHTML(ref(relation))``
``'<ul>' + pages.sort().map((p) => `<li>${dev.toHTML(ref(p))}</li>`).join('') + '</ul>'``
```

</p>
</details>

- Is it divided in 3 "sections", conceptually?
  - "Query"
  - "Arguments"
  - "Results"
- If so, could you provide another tutorial with breakdowns of "_Arguments_" and "_Results_"? 

Thank you very much

**Comment by stdword** [comment, marked as answer]
Hello! Thank you for a feedback  ❤️

> How to change the direction (vertical → horizontal) of displayed blocks?

I've just updated this template to support horizontal style layout. It is available [here](https://github.com/stdword/logseq13-full-house-plugin/discussions/21).

<img width="600px" src="https://github.com/stdword/logseq13-full-house-plugin/assets/1984175/6c9f7905-9ca7-4662-8533-c89537da6d7d"/>
<br/>
<br/>
<br/>

> How to rename the displayed title, as in "Pages with X containing..."?

<img width="800px" src="https://github.com/stdword/logseq13-full-house-plugin/assets/1984175/e24ffbed-1dc4-4ed4-bf9d-57221606e9a1"/>
<br/>
<br/>
<br/>

> Is it divided in 3 "sections", conceptually?

Now, thing get more complex:
<img width="800px" src="https://github.com/stdword/logseq13-full-house-plugin/assets/1984175/cb12b1ea-966b-4541-a1f7-8b1f99d8fb1d"/>
